NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# Ah view user 23

# Description

As an admin, I would like to see a list of all the User Profiles so that I can keep track of who is using the system.
Given the user is an admin in the Tabloid application
When they select the User Profiles menu option
Then they should be directed to the User Profiles list page
And each user in the list should display the full name, the display name and the user type.
And the list should be in ordered alphabetically by user display name



**Created the following files and methods:**

- GetAllUsers()
-UserProfileList.js



**Related Ticket(s)**
https://trello.com/c/St37rcAf

#23 View All User Profiles

**Type of Change**


# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `AH-ViewUser-23` as one of the available branches
- `git checkout AH-ViewUser-23` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin AH-ViewUser-23`
- `Start TabloidMVC.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)


- You will be presented with a login screen. Enter the email foo@bar.com and click "Login" (no password required)

- You will be presented with the welcome screen that simply says "hello"

- Click the “User Profiles” link in the topside header

-You should see a table that contains all users from your database as well as their corresponding Id, full name, display name, and user type. This table should be in alphabetical order by Last Name

-To the right of  each item on the list you should see an “Edit” button. This currently does nothing.


-Run this SQL query:

```
SELECT * FROM UserProfile;
```
 to confirm that the list matches users in the database.


# Checklist:
